command/thumbnails.php

```diff
@@ -0,0 +1,473 @@
+	 *
+	 * @return array
+	 * @throws ForbiddenCommandException|InputCommandException
+	 * @see OC\Files\Utils\Scanner::scan
+	 */
+	protected function performOperation($scanOperation, $user, $dir, OutputInterface $output
+	) {
+		if (!Filesystem::isValidPath($dir)) {
+			throw new InputCommandException('Invalid path to scan');
+		}
+		$mounts = $this->getMounts($dir, $user);
+		$result = [];
+		foreach ($mounts as $mount) {
+			$previewsAllowed = $mount->getOption('previews', true);
+			if (is_null($mount->getStorage()) || !$previewsAllowed) {
+				continue;
+			}
+			$storage = $mount->getStorage();
+			// if the home storage isn't writable then the scanner is run as the wrong user
+			if ($storage->instanceOfStorage('\OCP\Files\IHomeStorage') and
+				(!$storage->isCreatable('') or !$storage->isCreatable('files'))
+			) {
+				throw new ForbiddenCommandException(
+					"<error>Home storage for user $user not writable</error>\nMake sure you're running the scan command only as the user the web server runs as"
+				);
+			}
+			// We can now execute the wanted operation
+			$result[] = call_user_func_array(
+				[$this, $scanOperation],
+				[$storage, $mount, $dir, $user, $output]
+			);
+		}
+
+		return $result;
+	}
+
+	/**
+	 * Determines if we have a file
+	 *
+	 * @param Node $node
+	 *
+	 * @return bool
+	 */
+	protected function isFile(Node $node) {
+		$isFile = false;
+		// TODO: Find a more reliable way as we can't trust FileInfo
+		if ($node->getType() === 'file') {
+			$isFile = true;
+		}
+
+		return $isFile;
+	}
+
+	/**
+	 * Makes sure the folder or file exists in the user's filesystem
+	 *
+	 * @param string $path a full path or a relative path starting from the user's folder
+	 * @param string|null $user
+	 *
+	 * @return bool|Node
+	 *
+	 */
+	protected function getNode($path, $user = null) {
+		if ($user) {
+			$path = '/' . $user . '/' . $path;
+		}
+		if ($this->rootFolder->nodeExists($path)) {
+			return $this->rootFolder->get($path);
+		}
+
+		return false;
+	}
+
+	/**
+	 * @return bool
+	 */
+	protected function hasBeenInterrupted() {
+		$cancelled = false;
+		pcntl_signal_dispatch();
+		if ($this->interrupted) {
+			$cancelled = true;
+		}
+
+		return $cancelled;
+	}
+
+	/**
+	 * Asks for confirmation before proceeding
+	 *
+	 * @param InputInterface $input
+	 * @param string $question
+	 * @param OutputInterface $output
+	 *
+	 * @throws InputCommandException
+	 */
+	protected function getConfirmation(InputInterface $input, $question, OutputInterface $output) {
+		$helper = $this->getHelper('question');
+		$confirmationQuestion = new ConfirmationQuestion($question, false);
+
+		$quiet = false;
+		if ($output->isQuiet()) {
+			$quiet = true;
+			$output->setVerbosity($output::VERBOSITY_NORMAL);
+		}
+		if (!$helper->ask($input, $output, $confirmationQuestion)) {
+			throw new InputCommandException("Operation aborted");
+		}
+		if ($quiet) {
+			$output->setVerbosity($output::VERBOSITY_QUIET);
```

Comment:
if you change the `$output::VERBOSITY_*` stuff to `OutputInterface::VERBOSITY_*` it solves the issue with the check-code